tests/test_apis.py

```diff
@@ -0,0 +1,209 @@
+"""Test the MyTardis and ICD11 APIs"""
+
+# pylint: disable=redefined-outer-name,invalid-name,protected-access
+from typing import Any, Dict
+
+import mock
+import pytest
+import responses
+from mock import MagicMock
+from requests.models import HTTPError, MissingSchema
+
+from src.ingestion_targets.print_lab_genomics.ICD11_API_agent import ICD11ApiAgent
+from src.ingestion_targets.print_lab_genomics.print_crate_dataclasses import (
+    MedicalCondition,
+)
+from src.mt_api.api_consts import CONNECTION__HOSTNAME
+from src.mt_api.apiconfigs import AuthConfig, MyTardisRestAgent
+
+
+@responses.activate
+def test_mytardis_rest_agent_get(
+    test_metadata_response: Dict[str, Any], auth: AuthConfig, test_schema_namespace: str
+) -> None:
+    """Test a basic get request from the MyTardis API agent
+
+    Args:
+        test_metadata_response (Dict[str, Any]): a test response expected from the agent
+        auth (AuthConfig): a test authentication config
+        test_schema_namespace (str): a test namespace for schemas to construct the request
+    """
+    mt_rest_agent = MyTardisRestAgent(auth, CONNECTION__HOSTNAME, None, False)
+    schema_stub = "schema/?namespace="
+    url = mt_rest_agent.api_template + schema_stub + test_schema_namespace
+    responses.add(
+        responses.GET,
+        url,
+        status=200,
+        json=test_metadata_response,
+    )
+
+    response = mt_rest_agent.mytardis_api_request("GET", url)
+    assert isinstance(response.json(), dict)
+    assert response.ok is True
+    assert response.status_code == 200
+
+
+@responses.activate
+def test_mytardis_rest_agent_no_auth(
+    test_metadata_response: Dict[str, Any], test_schema_namespace: str
+) -> None:
+    """Test a request that does not need authorization
+
+    Args:
+        test_metadata_response (Dict[str, Any]): a test response expected from the agent
+        auth (AuthConfig): a test authentication config
+        test_schema_namespace (str): a test namespace for schemas to construct the request
+    """
+    do_not_use_auth = MagicMock()
+    mt_rest_agent = MyTardisRestAgent(
+        do_not_use_auth, CONNECTION__HOSTNAME, None, False
+    )
+    schema_stub = "schema/?namespace="
+    url = mt_rest_agent.api_template + schema_stub + test_schema_namespace
+    responses.add(
+        responses.GET,
+        url,
+        status=200,
+        json=test_metadata_response,
+    )
+
+    response = mt_rest_agent.no_auth_request("GET", url)
+    do_not_use_auth.assert_not_called()
+    assert isinstance(response.json(), dict)
```

Comment:
These three tests are mocked so asserting them is surplus to requirements I think.

Reply:
reduced just down to checking the auth is not used